File: educational-system-consumer-admin-9000/src/main/java/edu/hubu/consumeradmin/service/ICourseService.java
package edu.hubu.consumeradmin.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.admin.Course;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "admin-provider-service")
public interface ICourseService {

    @GetMapping("/course/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/course/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "courseId") String orderBy);

    @GetMapping("/course/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "courseId") String orderBy
    );

    @PostMapping("/course/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "courseId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.Course course
    );

    @PostMapping("/course/")
    CommonResult save(@RequestBody Course course);

    @PutMapping("/course/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody Course course);

    @DeleteMapping("/course/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/course/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/course/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/admin/ICourseService.kt
package edu.hubu.commons.service.admin

import edu.hubu.commons.model.admin.Course;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface ICourseService {
    fun save(t: Course): Course
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: Course): Course
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): Course
    fun selectAllOrderBy(order: String = "courseId"): List<Course>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "courseId"): List<Course>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "courseId", t: edu.hubu.commons.mybatis.model.Course): List<edu.hubu.commons.mybatis.model.Course>
}
